### ISSUE TITLE
Problem HttpRequestMessage dynamic property

### ISSUE THREAD

**Issue by sgom84** [OData/AspNetCoreOData#225]
Hello. I have a project with .net core 3.1 and odata 7.5.7. (It works)

I've updated to .net 5 and odata 8.0.0. 
When I start the execution, the call to GetEdmModel() on my OdataModelBuilder object throws this exception:

System.ArgumentException: 'Found more than one dynamic property container in type 'HttpRequestMessage'. Each open type must have at most one dynamic property container. 

I've seen that with .net 5 HttpRequestMessage has deprecated the property "Properties" (IDictionary<string,object?>) and added the property "Options" (HttpRequestOptions that implements interface IDictionary<string,object>).

Any idea on why this problem happens? (It happened also with release candidate 3)
Thanks

**Comment by xuzhg** [comment]
@sgom84 You can call "Ignore(...)" method to ignore the property.

**Comment by sgom84** [comment]
Thank You. It worked.

**Comment by widmarcel** [comment]
@sgom84 I have the same problem but I don't understand the solution. Can please provide some more information?

**Comment by sgom84** [comment]
When I prepare the odatamodelbuilder, I do the following:

builder.ComplexType<HttpRequestMessage>().Ignore(hrm => hrm.Properties);

**Comment by AntoBenil-Official** [comment]
Issue: Migration from .NET 3.1 to .NET 9.0

I have the same issue here. I migrated from .NET 3.1 to .NET 9.0.

Here is my model builder:

`var modelBuilder = new ODataConventionModelBuilder
{
    Namespace = "WebAPI",
    ContainerName = "DefaultContainer"
};

modelBuilder.ComplexType<HttpRequestMessage>().Ignore(hrm => hrm.Properties);

modelBuilder.Function("GetEntityData")
            .Returns<Task<IActionResult>>()
            .Parameter<string>("EntityName");

modelBuilder.Function("AddEntityData")
    .Returns<Task<IActionResult>>()
    .Parameter<string>("EntityName");

var putEntityData = modelBuilder.Function("UpdateEntityData")
    .Returns<Task<IActionResult>>();

putEntityData.Parameter<string>("EntityName");
putEntityData.Parameter<Guid>("EntityId");

var deleteEntityData = modelBuilder.Function("DeleteEntityData")
    .Returns<Task<IActionResult>>();

deleteEntityData.Parameter<string>("EntityName");
deleteEntityData.Parameter<Guid>("EntityId");`


And I'm getting the following error:

`System.ArgumentException: Found more than one dynamic property container in type 'HttpRequestMessage'. Each open type must have at most one dynamic property container. (Parameter 'propertyInfo')
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddDynamicPropertyDictionary(PropertyInfo propertyInfo)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapStructuralType(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapType(StructuralTypeConfiguration edmType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.AddEntityType(Type type)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity, Boolean containsTarget)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapStructuralType(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapType(StructuralTypeConfiguration edmType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.AddEntityType(Type type)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity, Boolean containsTarget)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapStructuralType(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapType(StructuralTypeConfiguration edmType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.AddEntityType(Type type)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity, Boolean containsTarget)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapStructuralType(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapType(StructuralTypeConfiguration edmType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.AddEntityType(Type type)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapDerivedTypes(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapStructuralType(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapType(StructuralTypeConfiguration edmType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.AddEntityType(Type type)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity, Boolean containsTarget)
   at Microsoft.OData.ModelBuilder.StructuralTypeConfiguration.AddNavigationProperty(PropertyInfo navigationProperty, EdmMultiplicity multiplicity)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapStructuralType(StructuralTypeConfiguration structuralType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapType(StructuralTypeConfiguration edmType)
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.MapTypes()
   at Microsoft.OData.ModelBuilder.ODataConventionModelBuilder.GetEdmModel()
   at Microsoft.AspNetCore.OData.Extensions.ActionDescriptorExtensions.GetEdmModel(ActionDescriptor actionDescriptor, HttpRequest request, Type entityClrType)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.GetModel(Type elementClrType, HttpRequest request, ActionDescriptor actionDescriptor)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.GetODataQueryContext(Object responseValue, IQueryable singleResultCollection, ControllerActionDescriptor actionDescriptor, HttpRequest request)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.GetModelBoundPageSize(ActionExecutedContext actionExecutedContext, Object responseValue, IQueryable singleResultCollection, ControllerActionDescriptor actionDescriptor, HttpRequest request)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.OnActionExecuted(ActionExecutedContext actionExecutedContext, Object responseValue, IQueryable singleResultCollection, ControllerActionDescriptor actionDescriptor, HttpRequest request)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.OnActionExecuted(ActionExecutedContext actionExecutedContext)
   at Microsoft.AspNetCore.Mvc.Filters.ActionFilterAttribute.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)`
